pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -76,4 +87,10 @@ stdenvNoCC.mkDerivation rec {
     platforms = [ "x86_64-linux" "x86_64-darwin" ];
     maintainers = [ maintainers.bhipple ];
   };
-}
+ } // stdenvNoCC.lib.optionalAttrs stdenvNoCC.isLinux {
+  # Since on Linux binaries are unmodified, they do not depend on stdenv
+	# and we can make them fixed-output derivations for cache efficiency.
```

Comment:
The reason why we can't have a fixed-output derivation is because we modify libraries by injecting a self-reference to "$out". The `stdenvNoCC`, `undmg` or `darwin.cctools` can and should give reproducible results, so they don't cause any fundamental problem.

Reply:
This comment is a cut and paste from the previous version, but OK, I'll modify it.